src/render/decorations/IHyprWindowDecoration.hpp

```diff
@@ -39,6 +39,12 @@ class IHyprWindowDecoration {
 
     virtual bool                     allowsInput();
 
+    virtual void                     onBeginDragOnDeco(const Vector2D&);
+
+    virtual bool                     onEndDragOnDeco(CWindow*, const Vector2D&);
```

Comment:
needs a comment what CWindow* and the return value is. For CWindow* a descriptive name would suffice.